### PR TITLE
Ensure that conflicts are staged and never ignored

### PR DESCRIPTION
**Pull request by ethomson**
Like https://github.com/libgit2/libgit2/pull/3139, LibGit2Sharp erroneously does not stage conflicted files, if the filenames match `.gitignore`.  Update stage to take conflict data into account.

### CODE REVIEW

**Comment by ethomson** [comment]
You know, some days I forget that we're building a tool that actively punishes Windows users for using two bytes to represent line endings instead of one.

**Comment by ethomson** [comment]
So after digging in here:  the `mergedrepo_wd` repository was created and has no explicit `core.autocrlf` settings.  The actual contents of the repository have CRLF checked in.  This (thankfully) matches the working directory copies.  This is akin to `core.autocrlf=false`.

**Comment by ethomson** [comment]
In case it's not clear, I favor a comprehensive solution to the tests issue at hand (the issue being that it's not super clear what the EOL settings are).  But this seems like a reasonable stopgap measure until we can be more comprehensive.

File: LibGit2Sharp/TreeEntryChanges.cs [hunk 1]
```diff
@@ -47,6 +49,11 @@ internal TreeEntryChanges(GitDiffDelta delta)
         public virtual ObjectId Oid { get; private set; }
 
         /// <summary>
+        /// The file exists (this is not a deletion.)
```
Comment (nulltoken) [inline LibGit2Sharp/TreeEntryChanges.cs]: Would there be a way to improve this documentation to help the user understand what we mean by "exists"?

I had to go through libgit2/libgit2#3139 in order to actually get it.

**Comment by nulltoken** [comment]
Would this fix #881?

**Comment by Therzok** [comment]
https://github.com/libgit2/libgit2sharp/commit/3cfb6f7fbc3c1a04d59cd1ae932efbf739df93ee Fixes #881.

File: LibGit2Sharp/TreeEntryChanges.cs (hunk 1 above)
Comment (ethomson) [inline LibGit2Sharp/TreeEntryChanges.cs]: I added additional documentation here, does this clarify this concept?

File: LibGit2Sharp/TreeEntryChanges.cs [hunk 2]
```diff
@@ -66,6 +79,16 @@ internal TreeEntryChanges(GitDiffDelta delta)
         /// </summary>
         public virtual ObjectId OldOid { get; private set; }
 
+        /// <summary>        
+        /// The file exists in the old side of the diff.
+        /// This is useful in determining if you have an ancestor
+        /// side to a conflict.  This will be flase during a
```
Comment (nulltoken) [inline LibGit2Sharp/TreeEntryChanges.cs]: s/flase/false/

File: LibGit2Sharp/TreeEntryChanges.cs [hunk 3]
```diff
@@ -66,6 +79,16 @@ internal TreeEntryChanges(GitDiffDelta delta)
...
+        /// conflcit that involves both the "ours" and "theirs"
```
Comment (nulltoken) [inline LibGit2Sharp/TreeEntryChanges.cs]: s/conflcit/conflict/

File: LibGit2Sharp/TreeEntryChanges.cs (hunk 2 above)
Comment (ethomson) [inline LibGit2Sharp/TreeEntryChanges.cs]: Fixed.

File: LibGit2Sharp/TreeEntryChanges.cs (hunk 3 above)
Comment (ethomson) [inline LibGit2Sharp/TreeEntryChanges.cs]: Fixed.

**Comment by nulltoken** [comment]
Awesome. Thanks!

**Pull request merged**